### ISSUE TITLE
Make defs_ field of Gringo::Defines  protected.

### ISSUE THREAD

**Issue by iensen** [potassco/clingo#185]
Hello,

Can we please make this field
https://github.com/potassco/clingo/blob/430fcdd96bf2f983e1aea7eb1b8fbe7c7f31e01e/libgringo/gringo/term.hh#L79
protected?

This seems to be the only reason I have to keep my fork now. I need a slightly different functionality for my prototype P-log solver (https://github.com/iensen/plog2.0) which is based on clingo.

**Comment by rkaminsk** [comment]
I would rather not make member variables protected. I am not sure why exactly you need to access this variable but have you noticed that the `Defines::defs()` member function already provides read-only access to this member? If you really need write access, then you can simply `const_cast` it.

**Comment by iensen** [comment]
Thank you! I am able to use the provided workaround.

I don't have a good answer to the why part:) I noticed that Clingo already has a nice class to handle the defines which includes, for example, cyclic definitions check, constant redefinition check, etc. So I wanted to  reuse it rather than inventing my own. However, I did not want to reuse the reporting part of it (frankly speaking, because I don't fully understand it). So I have hidden some methods in my own class and redefined error reporting. It is not a proper inheritance since Defines doesn't have virtual methods anyways. I will reconsider all this later.

I use clingo mostly because of its C++ API to find answer sets of some subprograms to compute values of non-random attributes of P-log. Using the defines is not essential, I just found it convenient.
